index.rst

```diff
@@ -2,38 +2,192 @@
+Check out the :doc:`documentation/tutorial`!
 
-* Transparent persistence for Python objects
-* Full ACID-compatible transaction support (including savepoints)
-* History/undo ability
-* Efficient support for binary large objects (BLOBs)
-* Pluggable storages
-* Scalable architecture
+Transactions
+============
 
+Transactions make programs a lot easier to reason about for three
+important reasons.
 
-Documentation
+Transactions are atomic
+  Changes made in a transaction are either saved in their entirety or
+  not at all.
+
+  This makes error handling a lot easier.  If you have an error, you
+  just abort the current transaction. You don't have to worry about
+  undoing previous database changes.
+
+Transactions provide isolation for managing concurrency
+  Transactions allow multiple logical threads (threads or processes)
+  to access databases and the database prevents the threads from
+  making conflicting changes.
+
+  This allows you to scale your application across multiple threads,
+  processes or machines without having to use low-level locking
+  primitives.
+
+  You still have to deal with concurrency on some level. For
+  timestamp-based systems like ZODB, you may have to retry conflicting
+  transactions. With locking-based systems, you have to deal with
+  possible deadlocks.
+
+Transactions effect multiple objects
+  Most NoSQL databases don't have transactions. Their notions of
+  consistency are much weaker, typically applying to single documents.
+  There can be good reasons to use NoSQL databases for their extreme
+  scalability, but otherwise, think hard about giving up the benefits
+  of transactions.
+
+ZODB transaction support:
+
+- `ACID <https://en.wikipedia.org/wiki/ACID>`_ transactions with
+  `snapshot isolation
+  <https://en.wikipedia.org/wiki/Snapshot_isolation>`_
+
+- Distributed transaction support using two-phase commit
+
+  This allows transactions to span multiple ZODB databases and to span
+  ZODB and non-ZODB databases.
+
+Other other notable ZODB features
+=================================
+
+Pluggable layered storage
+  ZODB has a pluggable storage architecture. This allows a variety of
+  storage schemes including memory-based, file-based and distributes
+  (client-server) storage.  Through storage layering, storage
+  components provide compression, encryption, replication and more.
+
+Database caching with invalidation
+  Every database connection has a cache that is a consistent partial database
+  replica. When accessing database objects, data already in the cache
+  is accessed without any database interactions.  When data are
+  modified, invalidations are sent to clients causing cached objects
+  to be invalidated. The next time invalidated objects are accessed
+  they'll be loaded from the database.
+
+  This makes caching extremely efficient, but provides some limit to
+  the number of clients.  The server has to send an invalidation
+  message to each client for each write.
+
```

Comment:
This only applies to some storages (e.g., ClientStorage) though. Other storages, notably RelStorage, use a different model for invalidation. Perhaps that's worth mentioning, since the last point was about the availability of different storages.